falcon/request_helpers.py

```diff
@@ -129,3 +134,271 @@ def readlines(self, hint=None):
         """
 
         return self._read(hint, self.stream.readlines)
+
+
+class ParamProxy(object):
+    """
+    Provide `get_*` methods for both query-string parameters and
+    request body parameters.
+    """
+
+    def __init__(self, request, source):
+        self.req = request
+        self.source = source
+
+    def get(self, name, required=False, store=None):
```

Comment:
I suggest we take the opportunity to have a `default` parameter. This pattern is very common and useful:

```python
# Means: return 'fallback' only if 'key' is not defined, else return value even if value is an empty string.
value = req.params.get('key', 'fallback')
# Here 'fallback' will be set even if 'key' exists but is an empty string.
value = req.params.get('param') or 'fallback'
```

I think it's important to distinguish between `key` exist but it's empty and `key` is not given. A use case is resource patching, one could pass an empty string to reset some property.

Drawback here is that ideally `default` should be the first kwarg to make the common pattern shorter (`req.params.get('x', 'y')` rather than `req.params.get('x', default='y')`), but then this would be API breaking change.